sdks/cpp/lite/include/NamedChoiceConstraint.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+/**
+ * @file NamedChoiceConstraint.h
+ * @brief A constraint that checks if a value is within a named choice
+ * @author isaac.robert@rossvideo.com
+ * @date 2024-07-25
+ * @copyright Copyright (c) 2024 Ross Video
+ */
+
+#include <common/include/IConstraint.h>
+#include <google/protobuf/message_lite.h>
+#include <lite/include/PolyglotText.h>
+
+using catena::common::IConstraint;
+using catena::lite::PolyglotText;
+
+/**
+ * @brief Named choice constraint, ensures a value is within a named choice.
+ */
+template <typename T> 
+class NamedChoiceConstraint : public IConstraint {
+public:
+    /**
+     * @brief Construct a new Named Choice Constraint object
+     * @param init the list of choices
+     */
+    NamedChoiceConstraint(std::vector<std::pair<T, PolyglotText::ListInitializer>> init,
+        bool strict, std::string oid, bool shared)
+        : strict_{strict}, default_{init[0].first} {
+        for (size_t i = 0; i < init.size(); ++i) {
+            choices_[init[i].first] = init[i].second;
+        }
+        setOid(oid);
+        setShared(shared);
+    }
+
+    /**
+     * @brief set the value in dst to the value in src if the value in src is valid 
+     * @param dst the value to write the constrained value to
+     * @param src the value to read and apply the constraint on
+     */
+    void apply(void* dst, void* src) const override {
+        auto& update = *reinterpret_cast<catena::Value*>(src);
+        
+        if constexpr(std::is_same<T, int32_t>::value) {
+            // ignore the request if src is not valid
+            if (!update.has_int32_value()) { return; }
+
+            // constrain the choice if strict 
+            if (strict_ && !choices_.contains(update.int32_value())) {
+                update.set_int32_value(default_);
+            }
+            reinterpret_cast<catena::Value*>(dst)->set_int32_value(update.int32_value());
```

Comment:
this is setting the value of one catena::Value object from another - I think the desired behavior is to set the value of the parameter to the src value `dst.fromProto(update);`

let's discuss